Examples/tootsdk-release/Sources/tootsdk-release/Posts/MakePost.swift

```diff
@@ -0,0 +1,179 @@
+        else {
+            print("unable to retrieve releases atom feed")
+            return nil
+        }
+        
+        let parser = FeedParser(data: data)
+        
+        let result = parser.parse()
+        
+        switch result {
+        case .success(let feed):
+            if let entry = feed.atomFeed?.entries?.first {
+                let tag = entry.title ?? ""
+                let url = entry.links?.first?.attributes?.href ?? ""
+                
+                let changes = try await parseChanges(html: entry.content?.value ?? "")
+                
+                let params = createPost(tag: tag, releaseURL: url, changes: changes)
+                return params
+            }
+        case .failure(let error):
+            print(error.localizedDescription)
+        }
+        
+        return nil
+    }
+    
+    private func parseChanges(html: String) async throws -> [String] {
+        let changes = html.split(separator: "<li>")
+        
+        var changesToReturn: [String] = []
+        
+        for change in changes {
+            if let changeLine = try await parseChange(change: String(change)) {
+                changesToReturn.append(changeLine)
+            }
+        }
+        
+        return changesToReturn
+    }
+    
+    private func parseChange(change: String) async throws -> String? {
+        let titleHTML = try SwiftSoup.parse(change).body()?.text() ?? ""
+        let title = titleHTML.split(separator: " by")
+        
+        if title.count > 1 {
+            let title = String(title.first ?? "")
+            let users = try await parseUsers(html: change)
+            
+            var userCredits: [String] = []
+            
+            for user in users {
+                if let mastodonURL = try await getUserMastodon(user: user) {
+                    userCredits.append(mastodonURL)
+                } else {
+                    userCredits.append("https://github.com/" + user)
+                }
+            }
+            
+            return title + " " + userCredits.joined(separator: ", ")
+        } else {
+            return nil
+        }
+    }
+    
+    private func parseUsers(html: String) async throws -> [String] {
+        let usersHTML = try SwiftSoup.parse(html).getElementsByClass("user-mention")
+        
+        let users = try usersHTML.map({ user in
+            try user.text().replacingOccurrences(of: "@", with: "")
+        })
+        
+        let unique = Array(Set(users))
+        
+        return unique
+    }
+    
+    private func getUserMastodon(user: String) async throws -> String? {
+        let mastodonURL = "https://api.github.com/users/" + user + "/social_accounts"
+        let (data, _ ) = try await urlSession.data(from: URL(string: mastodonURL)!)
+        
+        let socialAccounts = try JSONDecoder().decode([SocialAccount].self, from: data)
+        
+        if let userMastodonURL = socialAccounts.first(where: { $0.provider == "mastodon" })?.url {
```

Comment:
Note that Microsoft only supports mastodon, social links to other server types are not possible.

Reply:
Yeah. It’s frustrating but if they support others we can add to this